File: app/src/main/java/kh/edu/rupp/to_dolistapp/views/UserListFragment.java
package kh.edu.rupp.to_dolistapp.views;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.tabs.TabLayout;

import kh.edu.rupp.to_dolistapp.R;
import kh.edu.rupp.to_dolistapp.adapters.UserViewPagerAdapter;

public class UserListFragment extends Fragment {

    private TabLayout tabLayout;
    private ViewPager2 viewPager2;
    private UserViewPagerAdapter userViewPagerAdapter;
    private MaterialButton btnAdd;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_user_list, container, false);

        tabLayout = view.findViewById(R.id.tabLayout);
        viewPager2 = view.findViewById(R.id.viewPager);

        userViewPagerAdapter = new UserViewPagerAdapter(requireActivity());
        viewPager2.setAdapter(userViewPagerAdapter);

        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager2.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {}

            @Override
            public void onTabReselected(TabLayout.Tab tab) {}
        });

        viewPager2.registerOnPageChangeCallback(
                new ViewPager2.OnPageChangeCallback() {
                    @Override
                    public void onPageSelected(int position) {
                        super.onPageSelected(position);

                        TabLayout.Tab tab = tabLayout.getTabAt(position);

                        if (tab != null) {
                            tab.select();
                        }
                    }
                });

        return view;
    }
}

File: app/src/main/java/kh/edu/rupp/to_dolistapp/views/UserListFragment.kt
package kh.edu.rupp.to_dolistapp.views

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewpager2.widget.ViewPager2
import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback
import com.google.android.material.button.MaterialButton
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
import kh.edu.rupp.to_dolistapp.R
import kh.edu.rupp.to_dolistapp.adapters.UserViewPagerAdapter

class UserListFragment : Fragment() {
    private var tabLayout: TabLayout? = null
    private var viewPager2: ViewPager2? = null
    private var userViewPagerAdapter: UserViewPagerAdapter? = null
    private val btnAdd: MaterialButton? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_user_list, container, false)

        tabLayout = view.findViewById<TabLayout>(R.id.tabLayout)
        viewPager2 = view.findViewById<ViewPager2>(R.id.viewPager)

        userViewPagerAdapter = UserViewPagerAdapter(requireActivity())
        viewPager2!!.setAdapter(userViewPagerAdapter)

        tabLayout!!.addOnTabSelectedListener(object : OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                viewPager2!!.setCurrentItem(tab.getPosition())
            }

            override fun onTabUnselected(tab: TabLayout.Tab?) {}

            override fun onTabReselected(tab: TabLayout.Tab?) {}
        })

        viewPager2!!.registerOnPageChangeCallback(
            object : OnPageChangeCallback() {
                override fun onPageSelected(position: Int) {
                    super.onPageSelected(position)

                    val tab = tabLayout!!.getTabAt(position)

                    if (tab != null) {
                        tab.select()
                    }
                }
            })

        return view
    }
}